src/capabilities/capabilities.ts

```diff
@@ -49,11 +54,52 @@ export function fetchCapabilitiesOnFocus(): Promise<void> {
 }
 
 function resolveCapabilities(resolve: () => void): void {
-  void getCapabilities().then((capabilities) => {
-    capabilities?.enabled_features.forEach((feature) => {
-      enabledCapabilities[feature] = true;
-    });
-
+  void refreshCapabilities().then(() => {
     resolve();
+    startRefreshLoop();
+  });
+}
+
+const capabilitiesRefreshed = new EventEmitter<void>();
+
+export function onDidRefreshCapabilities(listener: () => void): Disposable {
```

Comment:
it is bit strange that the `listener` does not get the refreshed capabilities as input, isn't it?